src/core/statement.cpp

```diff
@@ -770,7 +770,7 @@ void statement_impl::describe()
         props.set_db_type(dbtype);
         props.set_data_type(backEnd_->to_data_type(dbtype));
 
-        switch (dbtype)
+        switch (backEnd_->exchange_dbtype_for(dbtype))
```

Comment:
I'm happy to accept suggestions for a better name for this function :grinning:

Reply:
I think the name is fine, but if you'd prefer, perhaps we should emphasize that this is used for dynamic/row-based API only by calling it `exchange_dbtype_for_row()` or `dynamic_exchange_dbtype_for()` perhaps?